internal/k8s/statusaddress.go

```diff
@@ -14,82 +14,82 @@
 package k8s
 
 import (
-	"context"
-
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	"github.com/projectcontour/contour/internal/annotation"
 	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/api/networking/v1beta1"
-	clientset "k8s.io/client-go/kubernetes"
 )
 
-// IngressStatusUpdater observes informer OnAdd events and
+// StatusAddressUpdater observes informer OnAdd and OnUpdate events and
 // updates the ingress.status.loadBalancer field on all Ingress
 // objects that match the ingress class (if used).
-type IngressStatusUpdater struct {
-	Client       clientset.Interface
-	Logger       logrus.FieldLogger
-	Status       v1.LoadBalancerStatus
-	IngressClass string
+type StatusAddressUpdater struct {
```

Comment:
Looking forward this field is going to be used for more than just status. We should take `Address` out of the name to allow it to update more than just the address.

Reply:
This one probably won't be used for anything else, it's only designed to be used to update `status.loadBalancer` stanzas in things. Other status updates will come from the Builder, replacing the `StatusClient` interface there.